server/bin/pbench-copy-sosreports.sh

```diff
@@ -109,31 +109,35 @@ for result in $list ;do
     # was located at different places in the hierarchy.
     sosreports=$(find $incoming/sysinfo $incoming/*/sysinfo $incoming/*/*/sysinfo -name 'sosreport*.tar.xz' 2>/dev/null)
     src=""
-    for x in $sosreports ;do
+    for x in ${sosreports}; do
         # sosreport should be RO
-        chmod 444 $x
-        if [ ! -f $x.md5 ] ;then
-            log_error "$TS: FAILED: no MD5 file found for $x"
-            nerrs=$nerrs+1
+        chmod 444 ${x}
+        if [[ -f ${x}.md5 ]]; then
+            suffix="md5"
+        elif [[ -f ${x}.sha256 ]]; then
+            suffix="sha256"
+        else
+            log_error "${TS}: FAILED: no MD5 or SHA256 file found for ${x}"
+            nerrs=${nerrs}+1
             continue
         fi
-        chmod 444 $x.md5
-        md5=$(md5sum $x)
-        md5=${md5%% *}
-        if [ "$md5" != $(cat $x.md5) ] ;then
-            log_error "$TS: FAILED: MD5 does not match for  $x"
-            nerrs=$nerrs+1
+        chmod 444 ${x}.${suffix}
+        val=$(${suffix}sum ${x})
+        val=${val%% *}
+        if [[ "${val}" != $(cat ${x}.${suffix}) ]]; then
```

Comment:
`md5sum` and `sha256sum` will do the check for you if you like, but, if we're going to be paranoid, we need to check that the checksum file we've got matches, so we could do something like this:
```
        if ! grep -q ${x} ${x}.${suffix} || ! ${suffix}sum --status -c ${x}.${suffix} ; then
```

Otherwise, you can use `$(< ${x}.${suffix})` instead of `$(cat ${x}.${suffix})`